src/data/battle-anims.ts

```diff
@@ -565,6 +586,12 @@ export function loadCommonAnimAssets(scene: BattleScene, startLoad?: boolean): P
   });
 }
 
+export function loadEncounterAnimAssets(scene: BattleScene, startLoad?: boolean): Promise<void> {
+  return new Promise(resolve => {
+    loadAnimAssets(scene, Array.from(encounterAnims.values()), startLoad).then(() => resolve());
+  });
+}
+
```

Comment:
Please use `async/await` where possible

```suggestion
export async function loadEncounterAnimAssets(scene: BattleScene, startLoad?: boolean): Promise<void> {
  await loadAnimAssets(scene, Array.from(encounterAnims.values()), startLoad);
}

```

Reply:
Also add some jsdoc